modules/gdscript/gdscript_parser.cpp

```diff
@@ -978,8 +978,11 @@ GDScriptParser::Node *GDScriptParser::_parse_expression(Node *p_parent, bool p_s
 						return NULL;
 					}
 					Node *n = _parse_expression(arr, p_static, p_allow_assign, p_parsing_constant);
-					if (!n)
+					if (!n) {
 						return NULL;
+					} else if (n->type == Node::TYPE_IDENTIFIER) {
+						_reduce_node_type(n);
```

Comment:
This shouldn't be called here. The type-check phase only happens after all parsing is finished. AFAIK this is already done: https://github.com/godotengine/godot/blob/bde52cc6881b3cb84602ffba1dede0a0180dbe4f/modules/gdscript/gdscript_parser.cpp#L6260-L6264